lib: os: cbprintf: -Wsign-compare warning in cbprintf_complete.c

### Describe the bug

A `-Wsign-compare` warning is emitted in `lib/os/cbprintf_complete.c` 
at line 1659 when building with `-Wall -Wextra` flags.

lib/os/cbprintf_complete.c:1659:51: warning: operand of '?:' changes 
signedness from 'sint_value_type' {aka 'long long int'} to 
'uint_value_type' {aka 'long long unsigned int'} due to unsignedness 
of other operand [-Wsign-compare]


### Regression

- [ ] This is a regression.

### Steps to reproduce

west build -p always -b qemu_x86 samples/hello_world -- -DEXTRA_CFLAGS="-Wall -Wextra"

### Relevant log output

```shell
/Users/gokul/Documents/zephyrproject/zephyr/lib/os/cbprintf_complete.c:1659:51: 
warning: operand of '?:' changes signedness from 
'sint_value_type' {aka 'long long int'} to 
'uint_value_type' {aka 'long long unsigned int'} 
due to unsignedness of other operand [-Wsign-compare]
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

**Zephyr version:v4.3.0
**Board: qemu_x86
**OS: macOS
**SDK version : 0.17.4

### Additional Context

_No response_

--- a/lib/os/cbprintf_complete.c
+++ b/lib/os/cbprintf_complete.c
@@ -1656,7 +1656,12 @@ int z_cbvprintf_impl(cbprintf_cb __out, void *ctx, const char *fp,
 			break;
 		case 'c':
 			bps = buf;
-			buf[0] = CHAR_IS_SIGNED ? value->sint : value->uint;
+			/* Use if-else to avoid -Wsign-compare warning */
+			if (CHAR_IS_SIGNED) {
+				buf[0] = value->sint;
+			} else {
+				buf[0] = value->uint;
+			}
 			bpe = buf + 1;
 			break;
 		case 'd':

